api/v1alpha1/servicebinding_types.go

```diff
@@ -59,6 +110,38 @@ type ServiceBindingList struct {
 	Items           []ServiceBinding `json:"items"`
 }
 
+// Application resource to inject the binding info.
+// It could be any process running within a container.
+type Application struct {
+	// API version of the referent.
+	// +optional
+	APIVersion string `json:"apiVersion"`
+
+	// Kind of the referent.
+	// +optional
+	Kind string `json:"kind"`
+
+	// Name of the referent.
+	// Mutually exclusive with Selector.
+	// +optional
+	Name string `json:"name,omitempty"`
```

Comment:
APIVersion and Kind should not be optional

```suggestion
	// API version of the referent.
	APIVersion string `json:"apiVersion"`

	// Kind of the referent.
	Kind string `json:"kind"`

	// Name of the referent.
	// Mutually exclusive with Selector.
	Name string `json:"name,omitempty"`
```